cf-keycrypt/cf-keycrypt.c

```diff
@@ -403,73 +433,153 @@ static bool CheckHeader(const char *key, const char *value)
             return true;
         }
     }
+    else if (StringSafeEqual(key, "Encrypted-for"))
+    {
+        /* TODO: do some verification that 'value' is valid hash digest? */
+        return true;
+    }
     else
     {
         Log(LOG_LEVEL_ERR, "Unsupported header: '%s'", key);
         return false;
     }
 }
 
-static bool ParseHeaders(FILE *input_file)
+/**
+ * Read from #input_file into #buffer at most #max_chars or until #delimiter is
+ * encountered. If #stop_on_space, also stop when ' ' is read. #buffer is
+ * NUL-terminated when this function returns.
+ *
+ * @warning #buffer has to be at least #max_chars+1 big to accomodate for the
+ *          terminating NUL byte.
+ */
+static inline bool ReadUntilDelim(FILE *input_file, char *buffer, size_t max_chars, char delimiter, bool stop_on_space)
 {
+    bool done = false;
+    size_t i = 0;
+    char c = fgetc(input_file);
+    while (!done && (i <= max_chars))
```

Comment:
Each iteration of the loop increments i by 1 and puts 1 byte into buffer. Then after the loop, 1 more byte is added to buffer (NUL). The less than or equal means this loop runs up to `max_chars + 1`times. The condition needs to be `i < max_chars`, to prevent 1 byte overflow.

Reply:
Yup, good catch! `max_chars` is the highest index in `char buffer[max_chars + 1]`.